### ISSUE TITLE
Support "Increase Contrast" iOS a11y setting in Material library.

### ISSUE THREAD

**Issue by rami-a** [flutter/flutter#62213]
## Use case

iOS users (and maybe one day Android users) that enable high contrast mode should be able to use Material Flutter apps with increased contrast.

## Proposal

Providing high contrast theme parameters to `MaterialApp`, similar to dark theme support. The cupertino library already supports this through `CupetinoDynamicColor`s.
